### PR TITLE
Addressing (#8797) issue

### PR DESCRIPTION
**Pull request by hovox**
…ct when file count is very large

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Addressing #8797 issue  
  [hovox](https://github.com/hovox)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/cocoapods/sandbox/pod_dir_cleaner.rb [hunk 1]
```diff
@@ -48,14 +48,21 @@ def path_list
       # @return [Array<Strings>] The paths that can be deleted.
       #
       def clean_paths
-        cached_used = used_files
+        cached_used = used_files.map do |path|
```
Comment (dnkoutso) [inline lib/cocoapods/sandbox/pod_dir_cleaner.rb]: map(&:downcase)

File: lib/cocoapods/sandbox/pod_dir_cleaner.rb [hunk 2]
```diff
@@ -48,14 +48,21 @@ def path_list
...
+          path.downcase
+        end
         glob_options = File::FNM_DOTMATCH | File::FNM_CASEFOLD
-        files = Pathname.glob(root + '**/*', glob_options).map(&:to_s)
+        files = Pathname.glob(root + "**/*", glob_options).map(&:to_s)
```
Comment (dnkoutso) [inline lib/cocoapods/sandbox/pod_dir_cleaner.rb]: Why double quotes?

File: lib/cocoapods/sandbox/pod_dir_cleaner.rb [hunk 3]
```diff
@@ -48,14 +48,21 @@ def path_list
...
+        # if file path exists in cached_used
+        # then we can find it in constant speed
+        # otherwise we will fallback to iterate on all cached_used files
+        # which will have linear speed
+        cached_used_set = cached_used.to_set
         files.reject do |candidate|
           candidate = candidate.downcase
-          candidate.end_with?('.', '..') || cached_used.any? do |path|
-            path = path.downcase
+          candidate.end_with?(".", "..") || cached_used_set.include?(candidate) || cached_used.any? do |path|
```
Comment (dnkoutso) [inline lib/cocoapods/sandbox/pod_dir_cleaner.rb]: Also here why double quotes?

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ## Master
 
+* Performance optimization for large number of files
+  [#8797](https://github.com/CocoaPods/CocoaPods/issues/8797)
```
Comment (dnkoutso) [inline CHANGELOG.md]: Please include your name

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ## Master
 
+* Performance optimization for large number of files
```
Comment (dnkoutso) [inline CHANGELOG.md]: Update description to refer about cleaning of the sandbox dir during installation

File: lib/cocoapods/sandbox/pod_dir_cleaner.rb [hunk 6]
```diff
@@ -48,14 +48,21 @@ def path_list
...
       def clean_paths
-        cached_used = used_files
+        cached_used = used_files.map do |path|
+          path.downcase
+        end
         glob_options = File::FNM_DOTMATCH | File::FNM_CASEFOLD
-        files = Pathname.glob(root + '**/*', glob_options).map(&:to_s)
+        files = Pathname.glob(root + "**/*", glob_options).map(&:to_s)
 
+        # create a set for optimization reasons
```
Comment (dnkoutso) [inline lib/cocoapods/sandbox/pod_dir_cleaner.rb]: can remove these comments. they are somewhat helpful but they are not needed.

File: lib/cocoapods/sandbox/pod_dir_cleaner.rb (hunk 2 above)
Comment (hovox) [reply lib/cocoapods/sandbox/pod_dir_cleaner.rb]: VS Code extension converting to double, will revert back :)

File: lib/cocoapods/sandbox/pod_dir_cleaner.rb (hunk 1 above)
Comment (hovox) [reply lib/cocoapods/sandbox/pod_dir_cleaner.rb]: done

File: lib/cocoapods/sandbox/pod_dir_cleaner.rb (hunk 3 above)
Comment (hovox) [reply lib/cocoapods/sandbox/pod_dir_cleaner.rb]: done

File: CHANGELOG.md (hunk 4 above)
Comment (hovox) [reply CHANGELOG.md]: done

File: CHANGELOG.md (hunk 5 above)
Comment (hovox) [reply CHANGELOG.md]: Not sure I have understood you correctly, please let me know if I need to change this line again.

File: lib/cocoapods/sandbox/pod_dir_cleaner.rb (hunk 6 above)
Comment (hovox) [reply lib/cocoapods/sandbox/pod_dir_cleaner.rb]: done

**Pull request merged**